src/backend/metal/src/device.rs

```diff
@@ -363,6 +372,8 @@ impl Device {
 
         // compile with options
         let mut compiler_options = msl::CompilerOptions::default();
+        compiler_options.enable_point_size_builtin = primitive_class == MTLPrimitiveTopologyClass::Point;
+        compiler_options.resolve_specialized_array_lengths = true;
```

Comment:
Is it fine for portability to have this requirement? Basically this means we can't allow specialization constants to define array length.

Reply:
This is only based on the defaults, so no functional change. I haven't looked into semantics, and we should definitely take advantage of it in later changes.